RadarSDK/Radar.m

```diff
@@ -420,22 +420,37 @@ + (void)startTripWithOptions:(RadarTripOptions *)options {
 }
 
 + (void)startTripWithOptions:(RadarTripOptions *)options completionHandler:(RadarTripCompletionHandler)completionHandler {
-    [[RadarAPIClient sharedInstance] updateTripWithOptions:options
-                                                    status:RadarTripStatusStarted
+    [self startTripWithOptions:options trackingOptions:nil completionHandler:completionHandler];
+}
+
++ (void)startTripWithOptions:(RadarTripOptions *)tripOptions
+             trackingOptions:(RadarTrackingOptions *)trackingOptions
+           completionHandler:(RadarTripCompletionHandler)completionHandler {
+    [[RadarAPIClient sharedInstance] createTripWithOptions:tripOptions
                                          completionHandler:^(RadarStatus status, RadarTrip *trip, NSArray<RadarEvent *> *events) {
-                                             if (status == RadarStatusSuccess) {
-                                                 [RadarSettings setTripOptions:options];
+        if (status == RadarStatusSuccess) {
+            [RadarSettings setTripOptions:tripOptions];
 
-                                                 // flush location update to generate events
-                                                 [[RadarLocationManager sharedInstance] getLocationWithCompletionHandler:nil];
-                                             }
+            if (Radar.isTracking) {
+                [RadarSettings setPreviousTrackingOptions:[RadarSettings trackingOptions]];
+            } else {
+                [RadarSettings removePreviousTrackingOptions];
+            }
 
-                                             if (completionHandler) {
-                                                 [RadarUtils runOnMainThread:^{
-                                                     completionHandler(status, trip, events);
-                                                 }];
-                                             }
-                                         }];
+            if (trackingOptions) {
+                [self startTrackingWithOptions:trackingOptions];
+            }
+
+            // flush location update to generate events
+            [[RadarLocationManager sharedInstance] getLocationWithCompletionHandler:nil];
+        }
+
+        if (completionHandler) {
+            [RadarUtils runOnMainThread:^{
+                completionHandler(status, trip, events);
+            }];
+        }
+    }];
```

Comment:
@corypisano @nickpatrick 

I want to call out two thing with this code block:
1) We're starting tracking and doing `getLocationWithCompletionHandler` in parallel.  Should be ok but - any concerns about hitting rate limits or race conditions with that?
2) We're calling `completionHandler` in parallel with both the calls to `startTracking` and `getLocationWithCompletionHandler`.  Should be ok since call to `createTrip` has completed, which is the expectation for calling the `completionHandler`.  Just wanted to raise awareness of that.

Reply:
1 - hm not concerned with race condition as we're waiting for start trip to return now, but rate limit will squint

for 2 - getLocationWithCompletionHandler is passed in nil so shouldn't be calling completionHandler